src/services/tripHistorySearch.js

```diff
@@ -0,0 +1,58 @@
+import { Sequelize } from 'sequelize';
+import db from '../models';
+
+
+
+
+const findTrip = async (res, query, location, offset, limit) => {
+    try{
+    const result = await db.Trip.findAndCountAll({
+        limit, offset,
+        where: { destination: location },
+        include: [{
+          model: db.TravelRequest,
+          where: query
+        }]
+  
+      })
+      res.json({ result })
+  
+    } catch (err) {
+      return res.status(401).json(err.message);
+    }
+
+};
+
+
+export const displayNumberOfTrips = async (res, query) => {
+    let resultSet1 = [];
+    
+        await db.TravelRequest.findAndCountAll({ where: query })
```

Comment:
@edgar929 could you reduce this function complexity, lenght and improve its readability?
It has lots of .thens and ifs besides async & await aren't adequately used.